article-analyzer/src/main/java/co/inventorsoft/academy/spring/notifier/NotificationManager.java

```diff
@@ -0,0 +1,31 @@
+package co.inventorsoft.academy.spring.notifier;
+
+import co.inventorsoft.academy.spring.model.User;
+import co.inventorsoft.academy.spring.notifier.impl.EmailNotificationService;
+import co.inventorsoft.academy.spring.notifier.impl.SlackNotificationService;
+import co.inventorsoft.academy.spring.service.UserService;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class NotificationManager {
+    private final UserService userService;
+    private final EmailNotificationService emailNotificationService;
+    private final SlackNotificationService slackNotificationService;
+
+    public void sendNotifications() {
+        List<User> users = userService.getUsers();
+        if (users != null) {
```

Comment:
can null be here?

Reply:
No, because in the UserRepository, if an exception is thrown, an empty collection is returned, not null, so checking for null is unnecessary here.

Reply:
fixed